lib/matplotlib/axes/_axes.py

```diff
@@ -753,6 +753,164 @@ def text(self, x, y, s, fontdict=None, **kwargs):
         self._add_text(t)
         return t
 
+    def label_lines(self, *args, **kwargs):
+        handles, labels, extra_args, kwargs = mlegend._parse_legend_args(
+            [self],
+            *args,
+            **kwargs)
+        if len(extra_args):
+            raise TypeError(
+                'label_lines only accepts two nonkeyword arguments')
+
+        self._line_labels = []
+
+        # TODO(omarchehab98): Should we avoid using a Line2D method from inside
+        #                     the Axes class?
+        #                     If it's okay, we should check if handle is an
+        #                     instance of Line2D.
+        def get_last_data(handle):
+            last_x = handle.get_xdata()[-1]
+            last_y = handle.get_ydata()[-1]
+            return last_x, last_y
+        # Get last data point for each handle
+        xys = list(map(get_last_data, handles))
+
+        # Find the largest last x value
+        data_maxx = max(map(lambda point: point[0], xys))
+        # TODO(omarchehab98): We do eventually sort `xys` by `y`.
```

Comment:
Can you to just make these numpy arrays and use `np.max()`?  This `map(lambda...` syntax is not very idiomatic to the rest of our code base, particularly for something that is just an array of numbers.